### ISSUE TITLE
how to implement (5.2)

### ISSUE THREAD

**Issue by diegoalmesp** [JosephSilber/bouncer#92]
Hello! first of all, great package!. This is the first time I'm implementing Authorization in Laravel, and have many doubts:

First, I have successfully installed the package and run the migrations (the roles are on the DB).

So, I have a RESTful controller and need to allow some users to "use" some parts of it, like: role "creator" can update an entire entity, but role "reviewer" and only update a specific part (which I plan to put in a different method).

My doubts are: where should I place this code: 
```php
Bouncer::allow($user)->to('create', Post::class); or
Bouncer::assign('admin')->to($user); ?
```
in the AppServiceProvider@boot too?, if so, should I pass (as the $user variable) an entity of the logged in user?. This is silly question, I know, but I used to make my own athorizations in a "cave-man" way so I'm pretty confused with this.

Basically I would like to understand how and where in my app should I implement this "allow" and "assign" methods.

Thanks for your time in advance!

**Comment by diegoalmesp** [comment]
I got it.
Thanks anyway and great package!

**Comment by joshuatomasgarcia** [comment]
How did you solve this @diegoalmesp? I'm wondering the same thing and have yet to get my head around it.

**Comment by diegoalmesp** [comment]
hi @joshuatomasgarcia , that code was placed into the migrations. Also, if you have a registration system, then you will use this code to create new users and assign them permissions (and put it, for instance, in the Model).

When the migration runs, the reference for this user permission will be placed in the database (bouncer's ones).

So, I created a new service provider (BouncerServiceProvider) and setted the permissions there for every role. I have something like this:

```php
Bouncer::seeder(function () {
            Bouncer::allow('administrator')->to([
                'create-io',
                'update-io',
                'update-state',
                'view-advertisers',
                'view-billto',
                'view-confidential'
            ]);

            Bouncer::allow('creator')->to([
                'create-io',
                'update-io',
                'update-state',
                'view-advertisers',
                'view-billto',
                'view-confidential'
            ]);

            Bouncer::allow('reviewer')->to('update-state');

            Bouncer::allow('customer')->to('update-state');
```

These permissions are meant for the view, to be used like this :

```php
@can ('create-io')
    {{-- html --}}
@endcan
```

For more advanced permission, you can add logic to the model directly.

Hope it helps!
